git_info.js

```diff
@@ -41,12 +49,21 @@ module.exports = {
     });
   },
 
-  calculateBlobId: function(content) {
-    var header  = 'blob ' + content.length + '\0';
-    var store   = header + content;
-    var shasum  = crypto.createHash('sha1');
-    shasum.update(store);
-    return shasum.digest("hex");
-  }
+  calculateBlobId: calculateBlobId,
+
+  blobId: function(path, content) {
+    var stdout, returnBlobId = null;
 
+    try {
+      stdout = childProcess.execSync("git hash-object " + path, { stdio: [0, "pipe", "ignore"] });
+    } catch (e) { }
```

Comment:
WDYT of adding a log here of "git hash-object failed. Falling back to manual calculation"?